### PR TITLE
Reload configmap

### LINKED ISSUE #17: Bug: Reloading configmap for KM
# General Description:
When the config file on the pod changes, KM should use the new configs. Referencing https://github.com/asobti/kube-monkey/blob/master/config/config.go#L46

## How to Diagnose Problem:

1. Have debugging and dry run turned on: 
```toml
[kubemonkey]
dry_run = true
run_hour = 8
start_hour = 10
end_hour = 16
blacklisted_namespaces = ["default","kube-system"]
[debug]
enabled = true
schedule_delay = 10
force_should_kill = true
schedule_immediate_kill = true
```
Have kube-monkey running and a test deployment/pod. ex:
```yaml
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: counter
  namespace: test-system
spec:
  template:
    metadata:
      labels:
        kube-monkey/enabled: enabled
        kube-monkey/identifier: monkey-victim-counter
        kube-monkey/mtbf: '1'
    spec:
      containers:
      - name: count
        image: busybox
        args: [/bin/sh, -c, 'i=0; while true; do echo "$i: $(date)"; i=$((i+1)); sleep 1; done']
```

2. Edit the config file directly with `kubectl edit cm/cm-config-map` or edit the config file locally and reload with `kubectl create configmap cm-config-map --from-file=config.toml=cm-config-map.toml -o yaml --dry-run | kubectl replace -f -`
```toml
[kubemonkey]
dry_run = false
run_hour = 8
start_hour = 10
end_hour = 16
blacklisted_namespaces = ["default","kube-system"]
[debug]
enabled = true
schedule_delay = 10
force_should_kill = true
schedule_immediate_kill = true
```

3. Check the counter pod logs and watch the kube-monkey pod config file to see when it gets changed
```
kubectl get pods --namespace=kube-system | grep -o '^kube-monkey[^[:space:]]*' | xargs -L 1 -I f watch kubectl exec -ti "f" cat /etc/kube-monkey/config.toml

kubectl get pods --namespace=test-system | grep -o '^counter[^[:space:]]*' | xargs -L 1 -I f watch kubectl logs "f" --namespace="test-system" --tail=20
```

## Expected behavior: 
KM should detect the config change, switch off dry-run and kill the counter pod.

```
kubectl get pods --namespace=test-system | grep -o '^counter[^[:space:]]*' | xargs -L 1 -I f watch kubectl logs "f" --namespace="test-system" --tail=20
```

### PR DESCRIPTION
**Pull request by huanwei**
1.Describe what this PR did

fix the watchconfig to suport k8s configmap reload when configmap was modified.

2.Does this pull request fix one issue?

fix #17

3.Describe how you did it

below is the investigation.
```
[root@k8s-master examples]# kubectl -n kube-system -owide get pods |grep monkey
kube-monkey-7b8987b4d8-9rmb2               1/1       Running   0          4h        192.168.169.181   k8s-node2
[root@k8s-master examples]# 
[root@k8s-master examples]# vi configmap.yaml 
[root@k8s-master examples]# vi deployment.yaml 
[root@k8s-master examples]# 
[root@k8s-master examples]# kubectl -n kube-system exec -it kube-monkey-7b8987b4d8-9rmb2 sh
# ls -al /etc/kube-monkey/
total 4
drwxrwxrwx.  3 root root   75 Apr  4 09:42 .
drwxr-xr-x. 43 root root 4096 Apr  4 09:42 ..
drwxr-xr-x.  2 root root   24 Apr  4 09:42 ..4984_04_04_09_42_57.493638710
lrwxrwxrwx.  1 root root   31 Apr  4 09:42 ..data -> ..4984_04_04_09_42_57.493638710
lrwxrwxrwx.  1 root root   18 Apr  4 09:42 config.toml -> ..data/config.toml
```
after configmap modified，look again：
```
[root@k8s-master examples]# kubectl -n kube-system exec -it kube-monkey-7b8987b4d8-9rmb2 sh
# cd /etc/kube-monkey
# ls -al
total 4
drwxrwxrwx.  3 root root   75 Apr  4 14:07 .
drwxr-xr-x. 43 root root 4096 Apr  4 09:42 ..
drwxr-xr-x.  2 root root   24 Apr  4 14:07 ..4984_04_04_14_07_49.707929309
lrwxrwxrwx.  1 root root   31 Apr  4 14:07 ..data -> ..4984_04_04_14_07_49.707929309
lrwxrwxrwx.  1 root root   18 Apr  4 09:42 config.toml -> ..data/config.toml
```

but the pod did not recognize the change, so I have to fix the code in viper.

4.Describe how to verify it
after kubectl replace -f configmap.yaml, tail the pod's log:
```
[root@k8s-master examples]# kubectl logs -f kube-monkey-7b8987b4d8-85w79 -n kube-system
I0405 16:14:58.667535       1 config.go:74] Successfully validated configs
I0405 16:14:58.668139       1 main.go:52] Starting kube-monkey with v logging level 5 and local log directory /var/log/kube-monkey
I0405 16:14:58.761436       1 kubemonkey.go:19] Debug mode detected!
I0405 16:14:58.761472       1 kubemonkey.go:20] Status Update: Generating next schedule in 10 sec
I0405 16:15:08.761799       1 schedule.go:54] Status Update: Generating schedule for terminations
I0405 16:15:08.952412       1 schedule.go:47] Status Update: 1 terminations scheduled today
I0405 16:15:08.952476       1 schedule.go:49] v1.DaemonSet chaos-victim-ds scheduled for termination at 04/05/2018 17:00:00 +0800 CST
I0405 16:15:08.952562       1 kubemonkey.go:63] Status Update: Waiting to run scheduled terminations.
	********** Today's schedule **********
	k8 Api Kind	Kind Name		Termination Time
	-----------	---------		----------------
	v1.DaemonSet	chaos-victim-ds		04/05/2018 17:00:00 +0800 CST
	********** End of schedule **********


I0405 16:17:19.387225       1 config.go:55] Config change detected

```

5.Special notes for reviews
None

### CODE REVIEW

**Comment by huanwei** [comment]
@asobti there's an issue reported from travis, we may need to change the vendor to fix this issue. Can you kindly advise?
```
config/config.go:8:2: cannot find package "github.com/huanwei/viper" in any of:
	/home/travis/gopath/src/github.com/asobti/kube-monkey/vendor/github.com/huanwei/viper (vendor tree)
	/home/travis/.gimme/versions/go1.9.linux.amd64/src/github.com/huanwei/viper (from $GOROOT)
	/home/travis/gopath/src/github.com/huanwei/viper (from $GOPATH)
make: *** [test] Error 1
```

File: Dockerfile [hunk 1]
```diff
@@ -6,3 +6,7 @@ RUN if (dpkg -l | grep -cq tzdata); then \
         apt-get update && apt-get install tzdata -y --no-install-recommends apt-utils; \
     fi
 COPY kube-monkey /kube-monkey
+
+# timezone
+RUN /bin/cp /usr/share/zoneinfo/Asia/Shanghai /etc/localtime \
```
Comment (Spellchaser) [inline Dockerfile]: Local Development, remove before PR

File: config/config.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ import (
 
 	"github.com/fsnotify/fsnotify"
 	"github.com/golang/glog"
-	"github.com/spf13/viper"
+	"github.com/huanwei/viper"
```
Comment (Spellchaser) [inline config/config.go]: Local Development, revert before PR

File: examples/configmap.yaml [hunk 3]
```diff
@@ -11,4 +11,4 @@
       start_hour = 10
       end_hour = 16
       blacklisted_namespaces = ["kube-system"]
-      whitelisted_namespaces = [""]
+      whitelisted_namespaces = ["chaos-test"]
```
Comment (Spellchaser) [inline examples/configmap.yaml]: Local Development, remove before PR

File: victims/factory/daemonsets/daemonsets.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
+package daemonsets
```
Comment (Spellchaser) [inline victims/factory/daemonsets/daemonsets.go]: Don't bundle two PRs into one. 
Each PR should not build upon the other. This PR should be isolated to fixing to reloading the config map.

**Comment by Spellchaser** [comment]
Ahh I see... it's rebuilt from https://github.com/spf13/viper/pull/415

**Comment by Spellchaser** [comment]
To be closed, fix should be upstream @asobti

**Pull request closed without merging**